### PR TITLE
[coqtop] [byte] [ocaml] Address OCaml 4.13 deprecations

### PR DESCRIPTION
**Pull request by ejgallego**
Some functions in `Toploop` / `Topdirs` have been deprecated in OCaml
4.13, patch is straightforward.

### CODE REVIEW

File: topbin/coqtop_byte_bin.ml [hunk 1]
```diff
@@ -17,18 +17,17 @@ let _ = CErrors.register_handler (function
...
-      | Toploop.Directive_none f -> f ()
-      | _ -> ()
-    end
-    with
-    | Not_found -> ()
+    match Toploop.get_directive "rectypes" with
+    | None -> ()
+    | Some (Toploop.Directive_none f) -> f ()
+    | Some _ ->
+      Format.eprintf "warning: rectypes directive has changed!!@\n%!"
```
Comment (Alizter) [inline topbin/coqtop_byte_bin.ml:26]: What's going on with the end of the sentence? Also perhaps:
```suggestion
      Format.eprintf "Warning: rectypes directive has changed!!@\n%!"
```

Comment (ejgallego) [reply topbin/coqtop_byte_bin.ml]: Thanks!

File: topbin/coqtop_byte_bin.ml [hunk 2]
```diff
@@ -16,19 +16,30 @@ let _ = CErrors.register_handler (function
...
-      | Toploop.Directive_none f -> f ()
-      | _ -> ()
-    end
-    with
-    | Not_found -> ()
+    match _get_directive "rectypes" with
+    | None -> ()
+    | Some (Toploop.Directive_none f) -> f ()
+    | Some _ ->
+      Format.eprintf "Warning: rectypes directive has changed!!@\n%!"
```
Comment (Alizter) [inline topbin/coqtop_byte_bin.ml:38]: My question still stands about the end of this sentence? Are 3 exclamations and whatever `@\n%` is necessary?

Comment (ejgallego) [reply topbin/coqtop_byte_bin.ml:38]: The control characters are indeed required, see format documentation, first is a new line, second one is flush (mandatory) . We could remove the exclamations from the message, however this is more like an anomaly so i dunno. IMHO it is fine as is.

Comment (ejgallego) [reply topbin/coqtop_byte_bin.ml:38]: A common mistake is to forget "%!" when using format, that may not flush the buffer!

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
@coqbot merge now

**Pull request merged**